src/Chapter3.hs

```diff
@@ -971,6 +1074,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek
+    = Monday
+    | Tuesday
+    | Wednesday
+    | Thursday
+    | Friday
+    | Saturday
+    | Sunday
+    deriving (Show, Enum, Eq, Ord)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+  | day == Saturday = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
  | day == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂

Reply:
Thanks for this information!